app/controllers/katello/api/v2/systems_controller.rb

```diff
@@ -143,8 +143,7 @@ def create
     param :host_collection_ids, Array, :desc => N_("Specify the host collections as an array")
     def update
       system_params = system_params(params)
-      @system.update_attributes!(system_params)
-      @system.refresh_subscriptions if system_params[:autoheal]
+      sync_task(::Actions::Katello::System::Update, @system, system_params)
       respond_for_update
```

Comment:
I thought if it failed in the run phase no exception would be raised.  I'll verify this.